src/core/thread/mle.cpp

```diff
@@ -1121,6 +1121,12 @@ otError Mle::GetAlocAddress(Ip6::Address &aAddress, uint16_t aAloc16) const
     return error;
 }
 
+otError Mle::GetRlocAddress(Ip6::Address &aAddress, uint16_t aRloc16) const
```

Comment:
How about using a unique `GetLocatorAddress()` for both ALOC and RLOC address construction?

Reply:
Will revise accordingly.

Reply:
Created new function `GetLocatorAddress`